upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
```

Comment:
It's inefficient to create a new `XMLHttpRequest` for every chunk. Consider reusing a single instance or using the Fetch API, which offers better features for handling requests.

Example:
```javascript
const xhr = new XMLHttpRequest();
xhr.open("POST", serverUrl, true);

function uploadNextChunk() {
  // Your logic here
}
```